File: chat-service/src/clients/logging.client.ts
import axios from "axios";
import { env } from "../config/env";
import { logger } from "../utils/logger";

export type LogEventRequest = {
  eventType: string;
  actorUserId?: number | null;
  referenceId?: string | null;
  message: string;
  metadata?: Record<string, unknown> | null;
  timestamp: string;
};

const client = axios.create({
  baseURL: env.LOGGING_SERVICE_URL,
  timeout: 3000
});

export const LoggingClient = {
  async logEvent(event: LogEventRequest): Promise<void> {
    try {
      await client.post("/v1/api/logs/internal/events", event);
    } catch (error) {
      logger.warn("Logging service failed", { error });
    }
  }
};

File: event_service/src/main/java/event_service/client/LoggingClient.java
package event_service.client;

import event_service.client.dto.LogEventRequest;
import org.springframework.cloud.openfeign.FeignClient;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;

@FeignClient(name = "${logging.service.name}")
public interface LoggingClient {

    @PostMapping("/v1/api/logs/internal/events")
    void logEvent(@RequestBody LogEventRequest request);
}
